lib/rubocop/cop/rails/require_dependency.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks for the usage of `require_dependency`.
+      # Applications running in zeitwerk mode should not use require_dependency
+      #
+      # @example
+      #   # bad
+      #   require_dependency 'some_lib'
+      class RequireDependency < Base
+        MSG = 'Applications running in zeitwerk mode should not use `require_dependency`.'
```

Comment:
Can you tweak the message?

```suggestion
        MSG = 'Do not use `require_dependency` with zeitwerk mode.'
```